osc-server/src/main/java/org/osc/core/broker/util/network/NetworkSettingsApi.java

```diff
@@ -20,132 +20,93 @@
-        if (bootProtocol.equals("static")) {
-
-            networkSettingsDto.setDhcp(false);
-            networkSettingsDto.setHostIpAddress(networkInterfaceConfig.getProperty(this.IPADDRESS));
-            networkSettingsDto.setHostSubnetMask(networkInterfaceConfig.getProperty(this.NETMASK));
-
-            Properties networkConfig = new Properties();
-            try {
-                networkConfig = FileUtil.loadProperties(this.NETWORK_CONFIG_FILE);
-            } catch (IOException e) {
-                log.error("Failed to load network settings", e);
-            }
-
-            networkSettingsDto.setHostDefaultGateway(networkConfig.getProperty(this.GATEWAY));
-            try {
-                String[] dns = getDNSSettings();
-                networkSettingsDto.setHostDnsServer1(dns[0]);
-                networkSettingsDto.setHostDnsServer2(dns[1]);
-            } catch (IOException e) {
-                log.error("Failed to load DNS settings", e);
-            }
-
-        } else {
-            // only return IP in DHCP mode
-            networkSettingsDto.setDhcp(true);
-            String hostIpAddress = null;
-            try {
-                hostIpAddress = NetworkUtil.getHostIpAddress();
-            } catch (SocketException | UnknownHostException e) {
-                log.error("Failed to get host and/or ip address", e);
-            }
+            hostIpAddress = NetworkUtil.getHostIpAddress();
             networkSettingsDto.setHostIpAddress(hostIpAddress);
-            networkSettingsDto.setHostSubnetMask("");
-            networkSettingsDto.setHostDefaultGateway("");
-            networkSettingsDto.setHostDnsServer1("");
-            networkSettingsDto.setHostDnsServer2("");
-        }
-
+            networkSettingsDto.setHostSubnetMask(getIPv4LocalNetMask());
+            networkSettingsDto.setHostDefaultGateway(getDefaultGateway());
+            String[] dns = getDNSSettings();
+            networkSettingsDto.setHostDnsServer1(dns[0]);
+            networkSettingsDto.setHostDnsServer2(dns[1]);
+
+        } catch (SocketException | UnknownHostException e) {
+            log.error("Failed to get host and/or ip address", e);
+        } catch (IOException e) {
+            log.error("Failed to get DNS Server(s)", e);
+		}
         return networkSettingsDto;
     }
 
     private String[] getDNSSettings() throws IOException {
-        String[] dns = { "", "" };
+        String[] dns = { "", "", "" };
+        int index = 0;
 
-        int i = 0;
         try (FileReader fileReader = new FileReader(this.NETWORK_RESOLV);
              Scanner in = new Scanner(fileReader)) {
             while (in.hasNextLine()) {
                 String[] tokens = in.nextLine().split(" ");
                 String s = tokens[0];
                 if (s.startsWith("name") && tokens.length > 1) {
-                    dns[i++] = tokens[1];
-                    if (i > 2) {
+                    dns[index++] = tokens[1];
```

Comment:
On ln 55 and 56 looks like we are only  using two and not three dns entries. Maybe rather than adding an extra "" on the array just change the index > 2 to index > 1. Also using a const value to hold the value 2, size the array with this const and use const - 1 in the loop below. This will avoid this type of bugs int the future.

Reply:
I am changing the index value from 2 to 1 but other suggestion i'm leaving it for time being.

Reply:
could you fix what Emanoel has indicated?

Reply:
Yes Sudhir. I am fixing index related problem but counter related changes i'm leaving for time being.

Reply:
you could just use a list of string and convert it to array at the end using https://docs.oracle.com/javase/6/docs/api/java/util/Collection.html#toArray

Would avoid index problems and keep method signature the same.